valkyrie/lib/valkyrie/storage_adapter.rb

```diff
@@ -45,6 +45,10 @@ def stream
         io
       end
 
+      def disk_path
+        Pathname.new(io.path)
+      end
```

Comment:
I went ahead and folded DiskFile into the File class because there was an error about File not having a `disk_path` method